src/mainloop.h

```diff
@@ -62,6 +62,8 @@ class Mainloop {
         assert(_initialized);
         return _instance;
     }
+    Mainloop(const Mainloop &) = delete;
+    Mainloop &operator=(const Mainloop &) = delete;
```

Comment:
I don't really understand the choice of moving the constructor to be public. Should rather continue to be private

Reply:
It should give you one compiler error less when trying to use the c-tor. But I'll revert the change as it gives no real benefit.
When taking a look at the changes by Auterion regarding the testability of the mainloop, they changed the ownership so that the singleton isn't really needed any more. Maybe I'll incorporate something similar when trying to write more tests. (And I don't like singletons anyways, because they make testing quite complicated.)